Release notes YDB Server 25.1.2 (#18929)
ydb-platform/ydb: ydb/docs/ru/core/changelog-server.md

* [Реализовано](https://github.com/ydb-platform/ydb/pull/12567) сохранение оптимистичных блокировок в памяти при плавном перезапуске даташардов, что должно уменьшить число ошибок ABORTED из-за потери блокировок при балансировке таблиц между узлами.
* [Реализована](https://github.com/ydb-platform/ydb/pull/12689) отмена волатильных транзакций со статусом ABORTED при плавном перезапуске даташардов.
* [Добавлена](https://github.com/ydb-platform/ydb/pull/6342) возможность удалить `NOT NULL`-ограничения на столбец в таблице с помощью запроса `ALTER TABLE ... ALTER COLUMN ... DROP NOT NULL`.
* [Добавлено](https://github.com/ydb-platform/ydb/pull/9168) ограничение в 100 000 на число одновременных запросов на создание сессий в сервисе координации.
* [Увеличено](https://github.com/ydb-platform/ydb/pull/14219) максимальное [число столбцов в первичном ключе](./concepts/limits-ydb.md?#schema-object) с 20 до 30.
* **_(Экспериментально)_** [Добавлены](https://github.com/ydb-platform/ydb/pull/14075) строгие проверки прав доступа, включаемые установкой следующих флагов:

>   * `enable_strict_acl_check`;

Reviewer: «Не позволять удалять пользователя, если есть права» и «Запретить возможность выдачи прав несуществующим пользователям».

# Список изменений {{ ydb-short-name }} Server

## Версия 25.1 {#25-1}

### Версия 25.1.1. {#25-1-1-}

Дата выхода: 2025.

#### Функциональность

* [Добавлена](https://github.com/ydb-platform/ydb/issues/11454) поддержка консистентной [асинхронной репликации](./concepts/async-replication.md).
* Добавлена поддержка параметризованного типа Decimal.
* Добавлена поддержка [автопартиционирования топиков в CDC](./concepts/cdc.md#topic-partitions) для строковых таблиц. Включается установкой флага `enable_topic_autopartitioning_for_cdc` в [конфигурации кластера](./maintenance/manual/dynamic-config#obnovlenie-dinamicheskoj-konfiguracii).
* [Добавлена](https://github.com/ydb-platform/ydb/pull/8264) возможность [изменить время хранения данных](./concepts/cdc.md#topic-options) в CDC топике с использованием выражения `ALTER TOPIC`.
* [Поддержан](https://github.com/ydb-platform/ydb/pull/7052) формат `DEBEZIUM_JSON` для потоков изменений (changefeed). Включается установкой флага `enable_changefeed_ debezium_json_format`.
* В операциях резервного копирования и восстановления расширен состав поддерживаемых объектов. Включается установкой флагов, указанных в скобках:

>   * [поддержка](https://github.com/ydb-platform/ydb/issues/7054) поток изменений (changefeed) (флаги `enable_changefeeds_export` и `enable_changefeeds_import`),

Reviewer 2: ```suggestion
  * [поддержка](https://github.com/ydb-platform/ydb/issues/7054) потока изменений (changefeed) (флаги `enable_changefeeds_export` и `enable_changefeeds_import`),
```

  * [поддержка](https://github.com/ydb-platform/ydb/issues/12724) представления (VIEW) (флаг `enable_view_export`).
* [Добавлена](https://github.com/ydb-platform/ydb/pull/12909) автоматическая проверка целостности резервных копий при импорте, которая предотвращает восстановление из поврежденных резервных копий и защищает от потери данных.
* [Добавлена](https://github.com/ydb-platform/ydb/pull/15570) возможность создания представлений (VIEW), использующих [UDF](./yql/reference/builtins/basic.md#udf) в своих запросах.
* Добавлены системные представления с информацией о [настройках прав доступа](./dev/system-views#auth) и [партициях строковых таблиц](./dev/system-views.md#partitions).
* Добавлены новые параметры в операторы [CREATE USER](./yql/reference/syntax/create-user.md) и [ALTER USER](./yql/reference/syntax/alter-user.md):
  * `HASH` - возможность задания пароля в зашифрованном виде,
  * `LOGIN` и `NOLOGIN` - разблокировка и блокировка пользователя.

> * Улучшена безопасность учетных записей:

Reviewer 2: ```suggestion
* Повышена безопасность учетных записей:
```

  * [Добавлена](https://github.com/ydb-platform/ydb/pull/11963) проверка сложности пароля пользователя.
  * [Реализована](https://github.com/ydb-platform/ydb/pull/12578) автоматическая блокировка пользователя при исчерпании лимита попыток ввода пароля.
  * [Добавлена](https://github.com/ydb-platform/ydb/pull/12983) возможность пользователю самостоятельно сменить свой пароль.

> * [Реализована](https://github.com/ydb-platform/ydb/issues/9748) возможность переключения функциональных флагов в рантайме. Большинство флагов будут применяться без необходимости рестарта.

Author: Добавила

* [Добавлена](https://github.com/ydb-platform/ydb/pull/6342) возможность удалить NOT NULL ограничения на столбец в таблице с помощью запроса `ALTER TABLE _ ALTER COLUMN _ DROP NOT NULL`.
* [Добавлено](https://github.com/ydb-platform/ydb/pull/9168) ограничение в 100 тысяч на число одновременных запросов на создание сессий в сервисе координации.
* [Увеличено](https://github.com/ydb-platform/ydb/pull/14219) максимальное [количество столбцов в первичном ключе](./concepts/limits-ydb.md?#schema-object) с 20 до 30.
* [Реализована](https://github.com/ydb-platform/ydb/pull/14460) дополнительная процедура многократной перезаписи удаленных данных для снижения риска их прочтения при непосредственном обращении к блочным устройствам средствами ОС.
* **_(Экспериментально)_** [Добавлены](https://github.com/ydb-platform/ydb/pull/14075) строгие проверки прав доступа, которые включаются установкой флагов `enable_strict_acl_check` и:
  * `enable_strict_user_management` включает строгие проверки для локальных пользователей,
  * `enable_database_admin` включает функции администратора базы данных,

>   * `enable_data_erasure` включает стирания удаленных данных с дисковых устройств.

Author: Да, обновила

* [Исправлены](https://github.com/ydb-platform/ydb/pull/9507) редкие зависания в момент первоначального сканирования [CDC](./dev/cdc.md).
* [Исправлена](https://github.com/ydb-platform/ydb/pull/11483) обработка незавершенных схемных транзакций в даташардах при перезапуске системы.
* [Исправлена](https://github.com/ydb-platform/ydb/pull/10460) ошибка несогласованного чтения из топика при попытке явно подтвердить сообщение, прочитанное в рамках транзакции. Теперь пользователь при попытке потвердить сообщение получит ошибку.
* [Исправлена](https://github.com/ydb-platform/ydb/pull/12220) ошибка, из-за которой автопартиционирование некорректно работало при работе с топиком в рамках транзакции.
* [Исправлены](https://github.com/ydb-platform/ydb/pull/12905) зависания транзакций при работе с топиками во время перезапуска таблеток.
* [Исправлена](https://github.com/ydb-platform/ydb/pull/13910) ошибка "Key is out of range" при импорте из S3-совместимого хранилища.
* [Исправлено](https://github.com/ydb-platform/ydb/pull/13741) некорректное определение конца поля с метаданными в [динамической конфигурации кластера](../maintenance/manual/dynamic-config).

> * [Улучшено](https://github.com/ydb-platform/ydb/pull/16420) построение вторичных индексов: при возникновении некоторых ошибок система теперь пытается восстановить процесс через перезапросы, а не прерывает его.

Author: Исправила

# Список изменений {{ ydb-short-name }} Server

## Версия 25.1 {#25-1}

### Версия 25.1.1. {#25-1-1-}

Дата выхода: 2025.

#### Функциональность

> * [Добавлена](https://github.com/ydb-platform/ydb/issues/11454) поддержка консистентной [асинхронной репликации](./concepts/async-replication).

Reviewer 3: `.md` в конце всех внутренних ссылок

* Добавлена поддержка параметризованного типа Decimal.

> * Добавлена поддержка [автопартиционирования топиков в CDC](./concepts/cdc#topic-partitions) для строковых таблиц. Включается установкой флага `enable_topic_autopartitioning_for_cdc` в [динамической конфигурации кластера](./maintenance/manual/dynamic-config#obnovlenie-dinamicheskoj-konfiguracii).

Reviewer 3: термин «динамическая конфигурация» как раз в 25.1 становится deprecated в пользу конфигурации V2

Author: А какой будет ссылка на статью про изменение флагов в конфиге?

#### Производительность

* [Добавлена](https://github.com/ydb-platform/ydb/pull/6509) поддержка [constant folding](https://en.wikipedia.org/wiki/Constant_folding) в оптимизаторе запросов по умолчанию, что повышает производительность запросов за счёт вычисления константных выражений на этапе компиляции.
* [Добавлен](https://github.com/ydb-platform/ydb/issues/6512) новый протокол гранулярного таймкаста, который позволит сократить времени выполнения распределенных транзакций (теперь замедление одного шарда не будет приводить к замедлению всех).
* [Реализована](https://github.com/ydb-platform/ydb/issues/11561) функциональность сохранения состояния даташардов в памяти при перезапусках. Это обеспечивает сокращение времени выполнения долгих транзакций благодаря уменьшению количества повторных попыток.
* Оптимизировано потребление памяти узлами хранения. - monster
* [Оптимизировано](https://github.com/ydb-platform/ydb/pull/10969) время запуска Hive.
* [Включена](https://github.com/ydb-platform/ydb/pull/6434) генерация расширенного формата полезной нагрузки.
* [Добавлена](https://github.com/ydb-platform/ydb/pull/6561) очередь этапов в репликации, которая сократила количество вычислений.

> * [Поддержан](https://github.com/ydb-platform/ydb/issues/12510) троттлинг входящей нагрузки на запись на VDisk, чтобы предотвратить аварийную ситуацию его 100% заполнения.

Author: Запросила информацию у автора изменений

* [Отключен](https://github.com/ydb-platform/ydb/pull/9491) заголовок больших двоичных объектов в VDisk.
* Улучшена диагностика и интроспекция ошибок с памятью.
* Уменьшено потребления памяти за счет очистки страниц аллокатора.
* Реализовано использование TEvWrite при записи для ускорения запросов.

#### Исправления ошибок

* [Исправлена](https://github.com/ydb-platform/ydb/pull/9707) ошибка в настройке [Interconnect](../concepts/glossary#actor-system-interconnect), приводящая к снижению производительности.
* [Ограничено](https://github.com/ydb-platform/ydb/pull/13993) одновременное удаление таблиц, чтобы избежать зависаний.

> * [Исправлена](https://github.com/ydb-platform/ydb/pull/9848) ошибка, которая возникала, если для системных таблеток одна нода указана несколько раз.

Reviewer 4: Она влияла на нас. В конфиге задублировалась нода и в таких условиях bootstrapper неправильно работал. И этот PR исправлял ровно эту ошибку. Там в PR даже ссылка на баг есть: https://github.com/ydb-platform/ydb/issues/9843

* [Исправлена](https://github.com/ydb-platform/ydb/pull/9723) ошибка чтения из асинхронных реплик.
* [Исправлены](https://github.com/ydb-platform/ydb/pull/9507) редкие зависания в момент первичного сканирования [CDC](./dev/cdc).
* [Исправлена](https://github.com/ydb-platform/ydb/pull/11483) обработка незавершенных схемных транзакций в даташардах при перезапуске системы.
* [Исправлены](https://github.com/ydb-platform/ydb/pull/18507) ошибки обработки опциональных столбцов и столбцов с типом UUID в строковых таблицах.
* [Исправлена](https://github.com/ydb-platform/ydb/pull/10460) ошибка из-за отправки сообщения о подтверждении обработки сообщения в транзакции, которая приводила к неконсистентному чтению из топика.
* [Исправлена](https://github.com/ydb-platform/ydb/pull/12220) ошибка, из-за которой автопартиционирование некорректно работало при работе с топиком в рамках транзакции.
* [Исправлены](https://github.com/ydb-platform/ydb/pull/12905) зависания транзакций при работе с топиками во время перезапуска таблеток. 
* [Исправлена](https://github.com/ydb-platform/ydb/pull/17122) редкая ошибка зависания клиентских приложений во время выполнения коммита.

> * [Исправлена](https://github.com/ydb-platform/ydb/pull/16649) ошибка, из-за которой distconf выдавал ошибку проверки и зависал при удалении узлов через устаревший интерфейс управления конфигурацией консоли.

Reviewer 5: Ну пока вряд ли кто-то им пользуется извне. Мы же ещё документацию толком не написали, как это всё через дистконф разворачивать.

* [Исправлена](https://github.com/ydb-platform/ydb/pull/13910) ошибка "Key is out of range" при импорте из S3-совместимого хранилища.
* [Исправлено](https://github.com/ydb-platform/ydb/pull/13741) некорректное определение конца поля с метаданными в [динамической конфигурации кластера](../maintenance/manual/dynamic-config).

> * [Добавлена](https://github.com/ydb-platform/ydb/pull/16420) ошибка Ydb::StatusIds::TIMEOUT в список, подлежащий ретраям, что улучшает возможность построения больших вторичных индексов.

Author: @azevaykin Саш, а построение насколько больших индексов завершалось таймаутом?

> * [Добавлена](https://github.com/ydb-platform/ydb/pull/16420) ошибка Ydb::StatusIds::TIMEOUT в список, подлежащий ретраям, что улучшает возможность построения больших вторичных индексов.

Reviewer: Индекс большой, в нём могу происходить ошибки. Раньше некоторые ошибки приводили к прерыванию построения целиком. Сейчас сделали так, что часть ошибок приводят к перезапросам и продолжению построения индекса.

* [Исправлена](https://github.com/ydb-platform/ydb/pull/16269) проблема зависания операции Drop Tablet в PQ tablet особенно во время задержек в работе Interconnect.
* https://github.com/ydb-platform/ydb/pull/9122 NodeBroker теперь отправляет дельты при регистрации нод в кластере.
* [Исправлена](https://github.com/ydb-platform/ydb/pull/16194) ошибка, возникавшая во время [компакшн](../concepts/glossary#compaction) VDisk.
* [Добавлена](https://github.com/ydb-platform/ydb/pull/15570) возможность создания представлений (VIEW), использующих [UDF](../yql/reference/builtins/basic#udf) в своих запросах.
* [Исправлена](https://github.com/ydb-platform/ydb/pull/15233) проблема, из-за которой длительные сессии чтения топика завершались с ошибками "too big inflight".
* [Исправлено](https://github.com/ydb-platform/ydb/pull/15515) зависание при чтении топика, если хотя бы одна партиция не имела входящих данных, но читалась несколькими потребителями.
* [Оптимизировано](https://github.com/ydb-platform/ydb/pull/18302) использование памяти в транзакциях с большим количеством участников за счет изменения механизма хранения и повторной отправки сообщений TEvReadSet.

> * [Исправлена](https://github.com/ydb-platform/ydb/pull/18270) ошибка репликации.

Reviewer 5: В предыдущих не должны были, это вроде в 25-1 только появилось.

> * [Исправлена](https://github.com/ydb-platform/ydb/pull/18625) ошибка, из-за которой неверно рассчитывалось использование памяти узлами.

Author: Саша говорит, что это не утечка, просто в hive не отправлялось число. 
Ошибка была исправлена в предыдущих версиях, поэтому из этих релиз ноутов убираем.

# Список изменений {{ ydb-short-name }} Server

## Версия 25.1 {#25-1}

### Версия 25.1.2.6 {#25-1-2-6}

Дата выхода: 2025.

> #### Функциональность

Reviewer 3: ❗Как обсуждали вчера в личке, конфигурация V2 потерялась.

* [Реализован](https://github.com/ydb-platform/ydb/pull/19504) [векторный индекс](./dev/vector-indexes.md) для приближенного векторного поиска. Включается установкой флага `enable_vector_index` в [конфигурации кластера](./maintenance/manual/dynamic-config.md) под гарантию неотката на предыдущие версии.
* [Добавлена](https://github.com/ydb-platform/ydb/issues/11454) поддержка консистентной [асинхронной репликации](./concepts/async-replication.md).
* Поддержаны `BATCH UPDATE` и `BATCH DELETE` выражения, позволяющие применить изменения больших строковых таблиц вне транзакционных ограничений.

> * Добавлена поддержка параметризованного [типа Decimal](./yql/reference/types/primitive.md#numeric).

Author: @azevaykin поможешь со ссылкой на PR?

> * Добавлена поддержка параметризованного [типа Decimal](./yql/reference/types/primitive.md#numeric).

Author: Старый PR, решили с Сашей не добавлять

* [Добавлена](https://github.com/ydb-platform/ydb/pull/8264) возможность [изменить время хранения данных](./concepts/cdc.md#topic-options) в CDC топике с использованием выражения `ALTER TOPIC`.
* [Поддержан](https://github.com/ydb-platform/ydb/pull/7052) [формат DEBEZIUM_JSON](./concepts/cdc.md#debezium-json-record-structure)  для потоков изменений (changefeed).
* [Добавлена](https://github.com/ydb-platform/ydb/pull/19507) возможность создавать потоки изменений (changefeed) к индексным таблицам.
* Добавлена возможность [указания числа реплик](./yql/reference/syntax/alter_table/indexes.md) для вторичного индекса. Включается установкой флага `enable_access_to_index_impl_tables` в конфигурации кластера.
* В операциях резервного копирования и восстановления расширен состав поддерживаемых объектов. Включается установкой флагов, указанных в скобках:
  * [поддержка](https://github.com/ydb-platform/ydb/issues/7054) потока изменений (changefeed) (флаги `enable_changefeeds_export` и `enable_changefeeds_import`),

>   * [поддержка](https://github.com/ydb-platform/ydb/issues/12724) представлений (VIEW) (флаг `enable_view_export`).
> * Добавлено автоудаление временных директорий и таблиц при экспорте в s3. Включается установкой флага `enable_export_auto_dropping` в конфигурации кластера.
> * [Добавлена](https://github.com/ydb-platform/ydb/pull/12909) автоматическая проверка целостности резервных копий при импорте, которая предотвращает восстановление из поврежденных резервных копий и защищает от потери данных.
> * [Добавлена](https://github.com/ydb-platform/ydb/pull/15570) возможность создания представлений, использующих [UDF](./yql/reference/builtins/basic.md#udf) в своих запросах.
> * Добавлены системные представления с информацией о [настройках прав доступа](./dev/system-views#auth), [истории перегруженных партиций](./dev/system-views#top-overload-partitions), [истории партиций строковых таблиц со сломанными блокировками (TLI)](./dev/system-views#top-tli-partitions).
> * Добавлены новые параметры в операторы [CREATE USER](./yql/reference/syntax/create-user.md) и [ALTER USER](./yql/reference/syntax/alter-user.md):
>   * `HASH` - возможность задания пароля в зашифрованном виде,
>   * `LOGIN` и `NOLOGIN` - разблокировка и блокировка пользователя.
> * Повышена безопасность учетных записей:
>   * [Добавлена](https://github.com/ydb-platform/ydb/pull/11963) проверка сложности пароля пользователя.
>   * [Реализована](https://github.com/ydb-platform/ydb/pull/12578) автоматическая блокировка пользователя при исчерпании лимита попыток ввода пароля.
>   * [Добавлена](https://github.com/ydb-platform/ydb/pull/12983) возможность пользователю самостоятельно сменить свой пароль.
> * [Реализована](https://github.com/ydb-platform/ydb/issues/9748) возможность переключения функциональных флагов в рантайме. Флаги, для которых в [proto файле](https://github.com/ydb-platform/ydb/blob/main/ydb/core/protos/feature_flags.proto#L60) не указан параметр `(RequireRestart) = true`, будут применяться без необходимости рестарта.
> * [Теперь](https://github.com/ydb-platform/ydb/pull/11329) самые старые (а не новые) локи меняются на полношардовые в случае превышения количества локов на шардах.

Reviewer 3: ```suggestion
  * [поддержка](https://github.com/ydb-platform/ydb/issues/12724) представлений (`VIEW`) (флаг `enable_view_export`).
* Добавлено автоудаление временных директорий и таблиц при экспорте в S3. Включается установкой флага `enable_export_auto_dropping` в конфигурации кластера.
* [Добавлена](https://github.com/ydb-platform/ydb/pull/12909) автоматическая проверка целостности резервных копий при импорте, предотвращающая восстановление из повреждённых резервных копий и защищающая от потери данных.
* [Добавлена](https://github.com/ydb-platform/ydb/pull/15570) возможность создания представлений, использующих [UDF](./yql/reference/builtins/basic.md#udf) в запросах.
* Добавлены системные представления с информацией о [настройках прав доступа](./dev/system-views#auth), [истории перегруженных партиций](./dev/system-views#top-overload-partitions) и [истории партиций строковых таблиц со сломанными блокировками (TLI)](./dev/system-views#top-tli-partitions).
* Добавлены новые параметры в операторы [CREATE USER](./yql/reference/syntax/create-user.md) и [ALTER USER](./yql/reference/syntax/alter-user.md):
  * `HASH` — возможность задания пароля в зашифрованном виде;
  * `LOGIN` и `NOLOGIN` — разблокировка и блокировка пользователя.
* Повышена безопасность учётных записей:
  * [Добавлена](https://github.com/ydb-platform/ydb/pull/11963) проверка сложности пароля пользователя;
  * [Реализована](https://github.com/ydb-platform/ydb/pull/12578) автоматическая блокировка пользователя при исчерпании лимита попыток ввода пароля;
  * [Добавлена](https://github.com/ydb-platform/ydb/pull/12983) возможность самостоятельной смены пароля пользователем.
* [Реализована](https://github.com/ydb-platform/ydb/issues/9748) возможность переключения функциональных флагов во время работы сервера {{ ydb-short-name }}. Флаги, для которых в [proto-файле](https://github.com/ydb-platform/ydb/blob/main/ydb/core/protos/feature_flags.proto#L60) не указан параметр `(RequireRestart) = true`, будут применяться без рестарта кластера.
* [Теперь](https://github.com/ydb-platform/ydb/pull/11329) самые старые (а не новые) блокировки меняются на полношардовые при превышении количества блокировок на шардах.
```

* [Реализовано](https://github.com/ydb-platform/ydb/pull/12567) сохранение оптимистичных блокировок в памяти в случае плавного перезапуска даташардов, что должно уменьшить количество ошибок ABORTED из-за потери блокировок при балансировке таблиц между узлами.
* [Реализована](https://github.com/ydb-platform/ydb/pull/12689) отмена волатильных транзакций со статусом ABORTED в случае плавного перезапуска даташардов.
* [Добавлена](https://github.com/ydb-platform/ydb/pull/6342) возможность удалить NOT NULL ограничения на столбец в таблице с помощью запроса `ALTER TABLE _ ALTER COLUMN _ DROP NOT NULL`.
* [Добавлено](https://github.com/ydb-platform/ydb/pull/9168) ограничение в 100 тысяч на число одновременных запросов на создание сессий в сервисе координации.
* [Увеличено](https://github.com/ydb-platform/ydb/pull/14219) максимальное [количество столбцов в первичном ключе](./concepts/limits-ydb.md?#schema-object) с 20 до 30.

> * **_(Экспериментально)_** [Добавлены](https://github.com/ydb-platform/ydb/pull/14075) строгие проверки прав доступа, которые включаются установкой флагов `enable_strict_acl_check` и:

Reviewer 3: Добавить пояснение что делает именно сам `enable_strict_acl_check` — у соседних оно есть, а этот остался без внимания.

> * [Реализовано](https://github.com/ydb-platform/ydb/pull/12567) сохранение оптимистичных блокировок в памяти в случае плавного перезапуска даташардов, что должно уменьшить количество ошибок ABORTED из-за потери блокировок при балансировке таблиц между узлами.
> * [Реализована](https://github.com/ydb-platform/ydb/pull/12689) отмена волатильных транзакций со статусом ABORTED в случае плавного перезапуска даташардов.
> * [Добавлена](https://github.com/ydb-platform/ydb/pull/6342) возможность удалить NOT NULL ограничения на столбец в таблице с помощью запроса `ALTER TABLE _ ALTER COLUMN _ DROP NOT NULL`.
> * [Добавлено](https://github.com/ydb-platform/ydb/pull/9168) ограничение в 100 тысяч на число одновременных запросов на создание сессий в сервисе координации.
> * [Увеличено](https://github.com/ydb-platform/ydb/pull/14219) максимальное [количество столбцов в первичном ключе](./concepts/limits-ydb.md?#schema-object) с 20 до 30.
> * **_(Экспериментально)_** [Добавлены](https://github.com/ydb-platform/ydb/pull/14075) строгие проверки прав доступа, которые включаются установкой флагов `enable_strict_acl_check` и:
>   * `enable_strict_user_management` включает строгие проверки для локальных пользователей,
>   * `enable_database_admin` включает функции администратора базы данных,
>   * `enable_data_erasure` [включает]((https://github.com/ydb-platform/ydb/pull/14460)) процедура многократной перезаписи удаленных данных для снижения риска их прочтения при непосредственном обращении к блочным устройствам средствами ОС.

Reviewer 3: ```suggestion
* [Реализовано](https://github.com/ydb-platform/ydb/pull/12567) сохранение оптимистичных блокировок в памяти при плавном перезапуске даташардов, что должно уменьшить число ошибок ABORTED из-за потери блокировок при балансировке таблиц между узлами.
* [Реализована](https://github.com/ydb-platform/ydb/pull/12689) отмена волатильных транзакций со статусом ABORTED при плавном перезапуске даташардов.
* [Добавлена](https://github.com/ydb-platform/ydb/pull/6342) возможность удалить `NOT NULL`-ограничения на столбец в таблице с помощью запроса `ALTER TABLE ... ALTER COLUMN ... DROP NOT NULL`.
* [Добавлено](https://github.com/ydb-platform/ydb/pull/9168) ограничение в 100 000 на число одновременных запросов на создание сессий в сервисе координации.
* [Увеличено](https://github.com/ydb-platform/ydb/pull/14219) максимальное [число столбцов в первичном ключе](./concepts/limits-ydb.md?#schema-object) с 20 до 30.
* **_(Экспериментально)_** [Добавлены](https://github.com/ydb-platform/ydb/pull/14075) строгие проверки прав доступа, включаемые установкой следующих флагов:
  * `enable_strict_acl_check`; 
  * `enable_strict_user_management` — строгие проверки для локальных пользователей;
  * `enable_database_admin` — функции администратора базы данных;
  * `enable_data_erasure` — включает процедуру многократной перезаписи удалённых данных для снижения риска их чтения при прямом обращении к блочным устройствам средствами операционной системы [(PR #14460)](https://github.com/ydb-platform/ydb/pull/14460).
```

>   * `enable_data_erasure` [включает]((https://github.com/ydb-platform/ydb/pull/14460)) процедура многократной перезаписи удаленных данных для снижения риска их прочтения при непосредственном обращении к блочным устройствам средствами ОС.

Reviewer 3: ‼️ Помимо происходящего переименования я слышал, что данная функциональность будет доступна не во всех редакциях {{ ydb-short-name }} — важно сразу правильно скоммуницировать это, чтобы у пользователей не было ощущения, что фичу сначала дали, а потом отбирают.

>   * `enable_data_erasure` [включает]((https://github.com/ydb-platform/ydb/pull/14460)) процедура многократной перезаписи удаленных данных для снижения риска их прочтения при непосредственном обращении к блочным устройствам средствами ОС.

Reviewer 3: Отсутствуют ссылки на документацию

#### Изменения с потерей обратной совместимости

> * Если вы используете [temporal over YDB](https://github.com/yandex/temporal-over-ydb), обновите его на версию [v1.23.0-ydb-compat](https://github.com/yandex/temporal-over-ydb/releases/tag/v1.23.0-ydb-compat) перед обновление YDB на текущую версию, чтобы избежать ошибок в выполнении запросов.

Reviewer 3: См. комментарий к английской версии

* [Реализована](https://github.com/ydb-platform/ydb/pull/15255) конвейерная обработка внутренних транзакций в [Node Broker](./concepts/glossary#node-broker-#node-broker), которая ускорила запуск динамических узлов в кластере YDB.
* [Улучшена](https://github.com/ydb-platform/ydb/pull/15607) устойчивость Node Broker к повышенной нагрузке со стороны узлов кластера.
* [Включены](https://github.com/ydb-platform/ydb/pull/19440) по умолчанию выгружаемые B-Tree индексы взамен невыгружаемых индексов SST, что позволяет сократить потребление памяти при хранении "холодных" данных.
* [Оптимизировано](https://github.com/ydb-platform/ydb/pull/15264) потребление памяти узлами хранения.
* [Оптимизировано](https://github.com/ydb-platform/ydb/pull/10969) время запуска Hive.
* [Оптимизирован](https://github.com/ydb-platform/ydb/pull/6561) процесс репликации.
* [Оптимизирован](https://github.com/ydb-platform/ydb/pull/9491) размер заголовка больших двоичных объектов в VDisk.

> * Улучшена диагностика и интроспекция ошибок с памятью.

Reviewer 3: Диагностика и интроспекция звучит как фича, а не производительность.

> #### Производительность
> 
> * [Добавлена](https://github.com/ydb-platform/ydb/pull/6509) поддержка [constant folding](https://en.wikipedia.org/wiki/Constant_folding) в оптимизаторе запросов по умолчанию, что повышает производительность запросов за счёт вычисления константных выражений на этапе компиляции.
> * [Добавлен](https://github.com/ydb-platform/ydb/issues/6512) новый протокол гранулярного таймкаста, который позволит сократить время выполнения распределенных транзакций (замедление одного шарда не будет приводить к замедлению всех).
> * [Реализована](https://github.com/ydb-platform/ydb/issues/11561) функциональность сохранения состояния даташардов в памяти при перезапусках, что позволяет сохранить блокировки и повысить шансы на успешное выполнение транзакций. Это обеспечивает сокращение времени выполнения долгих транзакций благодаря уменьшению количества повторных попыток.
> * [Реализована](https://github.com/ydb-platform/ydb/pull/15255) конвейерная обработка внутренних транзакций в [Node Broker](./concepts/glossary#node-broker-#node-broker), которая ускорила запуск динамических узлов в кластере YDB.
> * [Улучшена](https://github.com/ydb-platform/ydb/pull/15607) устойчивость Node Broker к повышенной нагрузке со стороны узлов кластера.
> * [Включены](https://github.com/ydb-platform/ydb/pull/19440) по умолчанию выгружаемые B-Tree индексы взамен невыгружаемых индексов SST, что позволяет сократить потребление памяти при хранении "холодных" данных.
> * [Оптимизировано](https://github.com/ydb-platform/ydb/pull/15264) потребление памяти узлами хранения.
> * [Оптимизировано](https://github.com/ydb-platform/ydb/pull/10969) время запуска Hive.
> * [Оптимизирован](https://github.com/ydb-platform/ydb/pull/6561) процесс репликации.
> * [Оптимизирован](https://github.com/ydb-platform/ydb/pull/9491) размер заголовка больших двоичных объектов в VDisk.
> * Улучшена диагностика и интроспекция ошибок с памятью.
> * Уменьшено потребления памяти за счет очистки страниц аллокатора.

Reviewer 3: ```suggestion
#### Производительность

* [Добавлена](https://github.com/ydb-platform/ydb/pull/6509) поддержка [свёртки констант](https://ru.wikipedia.org/wiki/%D0%A1%D0%B2%D1%91%D1%80%D1%82%D0%BA%D0%B0_%D0%BA%D0%BE%D0%BD%D1%81%D1%82%D0%B0%D0%BD%D1%82) в оптимизаторе запросов по умолчанию, что повышает производительность запросов за счёт вычисления константных выражений на этапе компиляции.
* [Добавлен](https://github.com/ydb-platform/ydb/issues/6512) новый протокол гранулярного таймкаста, который позволит сократить время выполнения распределённых транзакций (замедление одного шарда не будет приводить к замедлению всех).
* [Реализована](https://github.com/ydb-platform/ydb/issues/11561) функциональность сохранения состояния даташардов в памяти при перезапусках, что позволяет сохранить блокировки и повысить шансы на успешное выполнение транзакций. Это сокращает время выполнения длительных транзакций за счёт уменьшения числа повторных попыток.
* [Реализована](https://github.com/ydb-platform/ydb/pull/15255) конвейерная обработка внутренних транзакций в [Node Broker](./concepts/glossary.md#node-broker), что ускорило запуск динамических узлов в кластере {{ ydb-short-name }}.
* [Улучшена](https://github.com/ydb-platform/ydb/pull/15607) устойчивость Node Broker к повышенной нагрузке со стороны узлов кластера.
* [Включены](https://github.com/ydb-platform/ydb/pull/19440) по умолчанию выгружаемые B-Tree-индексы вместо невыгружаемых SST-индексов, что позволяет сократить потребление памяти при хранении «холодных» данных.
* [Оптимизировано](https://github.com/ydb-platform/ydb/pull/15264) потребление памяти узлами хранения.
* [Оптимизировано](https://github.com/ydb-platform/ydb/pull/10969) время запуска Hive.
* [Оптимизирован](https://github.com/ydb-platform/ydb/pull/6561) процесс репликации в распределённом хранилище.
* [Оптимизирован](https://github.com/ydb-platform/ydb/pull/9491) размер заголовка больших двоичных объектов в VDisk.
* Улучшена диагностика и интроспекция ошибок, связанных с памятью.
* Уменьшено потребление памяти за счёт очистки страниц аллокатора.
```

#### Исправления ошибок

* [Исправлена](https://github.com/ydb-platform/ydb/pull/9707) ошибка в настройке [Interconnect](../concepts/glossary#actor-system-interconnect), приводящая к снижению производительности.
* [Исправлена](https://github.com/ydb-platform/ydb/pull/13993) ошибка "Out of memory" при удалении очень больших таблиц за счет регулирования количества одновременно обрабатывающих данную операцию таблеток.
* [Исправлена](https://github.com/ydb-platform/ydb/pull/9848) ошибка, которая возникала, если в конфигурации для системных таблеток один узел базы данных указан несколько раз.
* [Устранена](https://github.com/ydb-platform/ydb/pull/11059) ошибка длительного (секунды) чтения данных при частых операциях перешардирования таблицы.

> * [Исправлена](https://github.com/ydb-platform/ydb/pull/9723) ошибка чтения из асинхронных реплик, которая приводила к сбою.

Reviewer 3: сбою чего?

> * [Исправлена](https://github.com/ydb-platform/ydb/pull/9707) ошибка в настройке [Interconnect](../concepts/glossary#actor-system-interconnect), приводящая к снижению производительности.
> * [Исправлена](https://github.com/ydb-platform/ydb/pull/13993) ошибка "Out of memory" при удалении очень больших таблиц за счет регулирования количества одновременно обрабатывающих данную операцию таблеток.
> * [Исправлена](https://github.com/ydb-platform/ydb/pull/9848) ошибка, которая возникала, если в конфигурации для системных таблеток один узел базы данных указан несколько раз.
> * [Устранена](https://github.com/ydb-platform/ydb/pull/11059) ошибка длительного (секунды) чтения данных при частых операциях перешардирования таблицы.
> * [Исправлена](https://github.com/ydb-platform/ydb/pull/9723) ошибка чтения из асинхронных реплик, которая приводила к сбою.
> * [Исправлены](https://github.com/ydb-platform/ydb/pull/9507) редкие зависания в момент первоначального сканирования [CDC](./dev/cdc.md).
> * [Исправлена](https://github.com/ydb-platform/ydb/pull/11483) обработка незавершенных схемных транзакций в даташардах при перезапуске системы.
> * [Исправлена](https://github.com/ydb-platform/ydb/pull/10460) ошибка несогласованного чтения из топика при попытке явно подтвердить сообщение, прочитанное в рамках транзакции. Теперь пользователь при попытке потвердить сообщение получит ошибку.
> * [Исправлена](https://github.com/ydb-platform/ydb/pull/12220) ошибка, из-за которой автопартиционирование некорректно работало при работе с топиком в рамках транзакции.
> * [Исправлены](https://github.com/ydb-platform/ydb/pull/12905) зависания транзакций при работе с топиками во время перезапуска таблеток.
> * [Исправлена](https://github.com/ydb-platform/ydb/pull/13910) ошибка "Key is out of range" при импорте из S3-совместимого хранилища.
> * [Исправлено](https://github.com/ydb-platform/ydb/pull/13741) некорректное определение конца поля с метаданными в [динамической конфигурации кластера](./maintenance/manual/dynamic-config).
> * [Улучшено](https://github.com/ydb-platform/ydb/pull/16420) построение вторичных индексов: при возникновении некоторых ошибок система ретраит процесс, а не прерывает его.

Reviewer 3: ```suggestion
* [Исправлена](https://github.com/ydb-platform/ydb/pull/9707) ошибка в настройке [Interconnect](./concepts/glossary.md#actor-system-interconnect), приводящая к снижению производительности.
* [Исправлена](https://github.com/ydb-platform/ydb/pull/13993) ошибка «Out of memory» при удалении очень больших таблиц за счёт регулирования числа одновременно обрабатывающих данную операцию таблеток.
* [Исправлена](https://github.com/ydb-platform/ydb/pull/9848) ошибка, возникавшая при указании одного и того же узла базы данных несколько раз в конфигурации для системных таблеток.
* [Устранена](https://github.com/ydb-platform/ydb/pull/11059) ошибка длительного (секунды) чтения данных при частых операциях перешардирования таблицы.
* [Исправлена](https://github.com/ydb-platform/ydb/pull/9723) ошибка чтения из асинхронных реплик, приводившая к сбою.
* [Исправлены](https://github.com/ydb-platform/ydb/pull/9507) редкие зависания при первоначальном сканировании [CDC](./dev/cdc.md).
* [Исправлена](https://github.com/ydb-platform/ydb/pull/11483) обработка незавершённых схемных транзакций в даташардах при перезапуске системы.
* [Исправлена](https://github.com/ydb-platform/ydb/pull/10460) ошибка несогласованного чтения из топика при попытке явно подтвердить сообщение, прочитанное в рамках транзакции. Теперь пользователь при попытке подтвердить сообщение получит ошибку.
* [Исправлена](https://github.com/ydb-platform/ydb/pull/12220) ошибка, из-за которой автопартиционирование некорректно работало при работе с топиком в транзакции.
* [Исправлены](https://github.com/ydb-platform/ydb/pull/12905) зависания транзакций при работе с топиками во время перезапуска таблеток.
* [Исправлена](https://github.com/ydb-platform/ydb/pull/13910) ошибка «Key is out of range» при импорте из S3-совместимого хранилища.
* [Исправлено](https://github.com/ydb-platform/ydb/pull/13741) некорректное определение конца поля с метаданными в [динамической конфигурации кластера](./maintenance/manual/dynamic-config.md).
* [Улучшено](https://github.com/ydb-platform/ydb/pull/16420) построение вторичных индексов: при возникновении некоторых ошибок система ретраит процесс, а не прерывает его.
```

* [Исправлена](https://github.com/ydb-platform/ydb/pull/16635) ошибка выполнения выражения `RETURNING` в операциях INSERT/UPSERT.
* [Исправлена](https://github.com/ydb-platform/ydb/pull/16269) проблема зависания операции "Drop Tablet" в PQ tablet особенно во время задержек в работе Interconnect.
* [Исправлена](https://github.com/ydb-platform/ydb/pull/16194) ошибка, возникавшая во время [компакшн](./concepts/glossary#compaction) VDisk.
* [Исправлена](https://github.com/ydb-platform/ydb/pull/15233) проблема, из-за которой длительные сессии чтения топика завершались с ошибками "too big inflight".
* [Исправлено](https://github.com/ydb-platform/ydb/pull/15515) зависание при чтении топика, если хотя бы одна партиция не имела входящих данных, но читалась несколькими потребителями.
* [Устранена](https://github.com/ydb-platform/ydb/pull/18614) редкая проблема перезагрузок PQ tablet.
* [Устранена](https://github.com/ydb-platform/ydb/pull/18378) проблема, при которой после обновления версии кластера Hive запускал подписчиков в дата-центрах без работающих узлов баз данных.

> * [Исправлена](https://github.com/ydb-platform/ydb/pull/19057) ошибка "Failed to set up listener on port 9092 errno# 98 (Address already in use)", возникающая при обновлении версии.

Reviewer 3: версии чего именно?

> * [Исправлена](https://github.com/ydb-platform/ydb/pull/16635) ошибка выполнения выражения `RETURNING` в операциях INSERT/UPSERT.
> * [Исправлена](https://github.com/ydb-platform/ydb/pull/16269) проблема зависания операции "Drop Tablet" в PQ tablet особенно во время задержек в работе Interconnect.
> * [Исправлена](https://github.com/ydb-platform/ydb/pull/16194) ошибка, возникавшая во время [компакшн](./concepts/glossary#compaction) VDisk.
> * [Исправлена](https://github.com/ydb-platform/ydb/pull/15233) проблема, из-за которой длительные сессии чтения топика завершались с ошибками "too big inflight".
> * [Исправлено](https://github.com/ydb-platform/ydb/pull/15515) зависание при чтении топика, если хотя бы одна партиция не имела входящих данных, но читалась несколькими потребителями.
> * [Устранена](https://github.com/ydb-platform/ydb/pull/18614) редкая проблема перезагрузок PQ tablet.
> * [Устранена](https://github.com/ydb-platform/ydb/pull/18378) проблема, при которой после обновления версии кластера Hive запускал подписчиков в дата-центрах без работающих узлов баз данных.
> * [Исправлена](https://github.com/ydb-platform/ydb/pull/19057) ошибка "Failed to set up listener on port 9092 errno# 98 (Address already in use)", возникающая при обновлении версии.
> * [Исправлена](https://github.com/ydb-platform/ydb/pull/18905) ошибка, приводившая к segmentation fault при одновременном выполнении запроса к healthcheck и отключении узла кластера.
> * [Исправлен](https://github.com/ydb-platform/ydb/pull/18899) сбой в [партиционировании строковой таблицы](./concepts/datamodel/table#partitioning_row_table) при выборе разделенного ключа из образцов доступа, содержащих смешанные операции с полным ключом и префиксом ключа (например, точное чтение или чтение диапазона).
> * [Исправлена](https://github.com/ydb-platform/ydb/pull/18647) [ошибка](https://github.com/ydb-platform/ydb/issues/17885), из-за которой тип индекса ошибочно определялся как GLOBAL SYNC, хотя в запросе явно указывался UNIQUE.
> * [Исправлена](https://github.com/ydb-platform/ydb/pull/16797) ошибка, из-за которой автопартиционирование топиков не работало, когда параметр конфигурации `max_active_partition` был задан с помощью выражения `ALTER TOPIC`.
> * [Исправлена](https://github.com/ydb-platform/ydb/pull/18938) ошибка, из-за которой `ydb scheme describe` возвращал список столбцов не в том порядке, в котором они были заданы при создании таблицы.

Reviewer 3: ```suggestion
* [Исправлена](https://github.com/ydb-platform/ydb/pull/16635) ошибка выполнения выражения `RETURNING` в запросах `INSERT INTO` и `UPSERT INTO`.
* [Исправлена](https://github.com/ydb-platform/ydb/pull/16269) проблема зависания операции «Drop Tablet» в PQ tablet, особенно во время задержек в работе Interconnect.
* [Исправлена](https://github.com/ydb-platform/ydb/pull/16194) ошибка, возникавшая во время [компакшна](./concepts/glossary.md#compaction) VDisk.
* [Исправлена](https://github.com/ydb-platform/ydb/pull/15233) проблема, из-за которой длительные сессии чтения топика завершались с ошибками «too big inflight».
* [Исправлено](https://github.com/ydb-platform/ydb/pull/15515) зависание при чтении топика, если хотя бы одна партиция не имела входящих данных, но читалась несколькими потребителями.
* [Устранена](https://github.com/ydb-platform/ydb/pull/18614) редкая проблема перезагрузок PQ tablet.
* [Устранена](https://github.com/ydb-platform/ydb/pull/18378) проблема, при которой после обновления версии кластера Hive запускались подписчики в датацентрах без работающих узлов баз данных.
* [Исправлена](https://github.com/ydb-platform/ydb/pull/19057) ошибка `Failed to set up listener on port 9092 errno# 98 (Address already in use)`, возникавшая при обновлении версии.
* [Исправлена](https://github.com/ydb-platform/ydb/pull/18905) ошибка, приводившая к segmentation fault при одновременном выполнении запроса к healthcheck и отключении узла кластера.
* [Исправлен](https://github.com/ydb-platform/ydb/pull/18899) сбой в [партиционировании строковой таблицы](./concepts/datamodel/table.md#partitioning_row_table) при выборе разделённого ключа из образцов доступа, содержащих смешанные операции с полным ключом и префиксом ключа (например, точное чтение или чтение диапазона).
* [Исправлена](https://github.com/ydb-platform/ydb/pull/18647) [ошибка](https://github.com/ydb-platform/ydb/issues/17885), из-за которой тип индекса ошибочно определялся как `GLOBAL SYNC`, хотя в запросе явно указывался `UNIQUE`.
* [Исправлена](https://github.com/ydb-platform/ydb/pull/16797) ошибка, из-за которой автопартиционирование топиков не работало, когда параметр конфигурации `max_active_partition` задавался с помощью выражения `ALTER TOPIC`.
* [Исправлена](https://github.com/ydb-platform/ydb/pull/18938) ошибка, из-за которой `ydb scheme describe` возвращал список столбцов не в том порядке, в котором они были заданы при создании таблицы.
```

  * `enable_data_erasure` [включает]((https://github.com/ydb-platform/ydb/pull/14460)) процедура многократной перезаписи удаленных данных для снижения риска их прочтения при непосредственном обращении к блочным устройствам средствами ОС.

#### YDB UI

* В редактор запросов [добавлена](https://github.com/ydb-platform/ydb-embedded-ui/pull/1974) поддержка частичной загрузки результатов — отображение начинается сразу при получении первого фрагмента с сервера без ожидания полного завершения запроса. Это позволяет быстрее получать результаты.
* [Улучшена](https://github.com/ydb-platform/ydb-embedded-ui/pull/1967) безопасность: элементы управления, которые недоступны пользователю, теперь не отображаются в интерфейсе. Пользователи не будут сталкиваться с ошибками "Доступ запрещен".

> * [Добавлен](https://github.com/ydb-platform/ydb-embedded-ui/pull/1981) добавлен поиск по идентификатору таблетки на вкладку "Tablets".

Reviewer 2: "добавлен добавлен " получилось

> * Добавлена подсказка по горячим клавишам, которая открывается по комбинации `⌘+K`.

Reviewer 2: а для Windows ?
